consensus/sync/sync_test.go

```diff
@@ -0,0 +1,159 @@
+package sync_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/consensus/driver"
+	"github.com/NethermindEth/juno/consensus/mocks"
+	"github.com/NethermindEth/juno/consensus/p2p"
+	"github.com/NethermindEth/juno/consensus/proposal"
+	"github.com/NethermindEth/juno/consensus/starknet"
+	consensusSync "github.com/NethermindEth/juno/consensus/sync"
+	"github.com/NethermindEth/juno/consensus/tendermint"
+	"github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/p2p/sync"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+var (
+	comittedHeight = -1
+	blockID        = uint64(9)
+)
+
+type (
+	listeners    = p2p.Listeners[starknet.Value, starknet.Hash, starknet.Address]
+	broadcasters = p2p.Broadcasters[starknet.Value, starknet.Hash, starknet.Address]
+)
+
+func mockTimeoutFn(step types.Step, round types.Round) time.Duration {
+	return 10 * time.Second
+}
+
+func newDB(t *testing.T) *mocks.MockTendermintDB[starknet.Value, starknet.Hash, starknet.Address] {
+	t.Helper()
+	ctrl := gomock.NewController(t)
+	// Ignore WAL for tests that use this
+	db := mocks.NewMockTendermintDB[starknet.Value, starknet.Hash, starknet.Address](ctrl)
+	db.EXPECT().GetWALEntries(gomock.Any()).AnyTimes()
+	db.EXPECT().SetWALEntry(gomock.Any()).AnyTimes()
+	db.EXPECT().Flush().AnyTimes()
+	db.EXPECT().DeleteWALEntries(gomock.Any()).AnyTimes()
+	return db
+}
+
+func TestSync(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	ctx, cancel := context.WithCancel(t.Context())
+
+	nodeAddr := starknet.Address(felt.FromUint64(123))
+	logger := utils.NewNopZapLogger()
+	tendermintDB := newDB(t)
```

Comment:
Why don't we just use the in memory version as suggested in the previous comment? It's simpler, shorter and reflects the behaviour better than the mock version?